src/+matlode/+rk/+erk/ERK.m

```diff
@@ -178,13 +161,13 @@
                     
                     %set new step to be in range
                     if stepController.Adaptive
-                        hN = min(hmax, hN);
+                        hN = min(hmax * tdir, hN * tdir) * tdir;
```

Comment:
I would honestly just prefer

`hN = tdir*min(abs(hmax), abs(hN));`

It is less dependent on you keeping track of where you did and did not multiply by the direction.